lib/mongo/bulk_write.rb

```diff
@@ -327,20 +327,22 @@ def can_hint?(connection)
     # @raise [ Error::InvalidUpdateDocument, Error::InvalidReplacementDocument ]
     #   if the document is invalid.
     def validate_requests!
-      @requests.each do |req|
-        if op = req.keys.first
-          if [:update_one, :update_many].include?(op)
-            if doc = maybe_first(req.dig(op, :update))
-              if key = doc.keys&.first
-                unless key.to_s.start_with?("$")
-                  raise Error::InvalidUpdateDocument.new(key: key)
+      if Mongo.validate_update_replace
```

Comment:
I would return if the value is false to keep the nesting in the rest of the method down.

Reply:
Actually, shouldn't this produce a warning if the flag is off? And I think in this case the warning would need to be throttled otherwise it can spam the logs.

Reply:
Mongoid does https://github.com/mongodb/mongoid/blob/master/lib/mongoid/query_cache.rb#L82, the flag can perhaps be stored in a new module or in `Config` if everything related to it is marked `@api private`.